tests/test_rand.py

```diff
@@ -96,63 +112,77 @@ def test_status(self):
         # It's hard to know what it is actually going to return.  Different
         # OpenSSL random engines decide differently whether they have enough
         # entropy or not.
-        self.assertTrue(rand.status() in (1, 2))
+        assert rand.status() in (True, False)
 
-    def test_egd_warning(self):
+    @pytest.mark.parametrize('args', [
+        (b"foo", 255),
+        (b"foo",),
+    ])
+    def test_egd_warning(self, args):
         """
         Calling egd raises :exc:`DeprecationWarning`.
         """
-        pytest.deprecated_call(rand.egd, b"foo", 255)
-        pytest.deprecated_call(rand.egd, b"foo")
+        pytest.deprecated_call(rand.egd, *args)
 
-    def test_egd_wrong_args(self):
+    @pytest.mark.parametrize('args', [
+        (),
+        (None,),
+        ("foo", None),
+        (None, 3),
+        ("foo", 3, None),
+    ])
+    def test_egd_wrong_args(self, args):
         """
         :meth:`OpenSSL.rand.egd` raises :exc:`TypeError` when called with the
         wrong number of arguments or with arguments not of type :obj:`str` and
         :obj:`int`.
         """
-        for args in [(),
-                     (None,),
-                     ("foo", None),
-                     (None, 3),
-                     ("foo", 3, None)]:
-            with pytest.raises(TypeError):
-                rand.egd(*args)
+        with pytest.raises(TypeError):
+            rand.egd(*args)
 
     def test_cleanup_wrong_args(self):
```

Comment:
Uh this whole test can go.  All those args tests are from times when pyOpenSSL was written in C but many of them don’t make a lot of sense anymore.  This one particularly.   We don’t have to test whether a function that takes no arguments raises a TypeError.  Let’s leave that to Python core. ;)